src/index.ts

```diff
@@ -166,6 +166,11 @@ export class WatsonMiddleware {
 
       const watsonRequest = await this.before(message, payload);
       let watsonResponse = await postMessage(this.conversation, watsonRequest);
+      if (typeof watsonResponse.output.error === 'string') {
+        throw new Error(
+          'Watson Assistant returned error: ' + watsonResponse.output.error,
+        );
```

Comment:
@germanattanasio Would it be better to set `message.watsonError = watsonResponse.output.error;` here and let `after` hook run?

Reply:
Yes

Reply:
I changed the code.